subprojects/plugins/src/main/java/org/gradle/api/tasks/GroovyRuntime.java

```diff
@@ -70,9 +73,11 @@
         "groovy-templates", "groovy-json", "groovy-xml", "groovy-groovydoc");
 
     private final Project project;
+    private final JvmEcosystemUtilities jvmEcosystemUtilities;
 
     public GroovyRuntime(Project project) {
         this.project = project;
+        this.jvmEcosystemUtilities = ((ProjectInternal) project).getServices().get(JvmEcosystemUtilities.class);
```

Comment:
Injecting this service via constructor is possible, but this class is in the public API and changing this constructor breaks binary compatibility.

Reply:
Does it work if you inject it via a method?
```java
@Inject
public JvmEcosystemUtilities getJvmEcosystemUtilities() {
    throw new UnsupportedOperationException();
}
```